src/side-loaded/basic-sideload/sideloaded.run.ts

```diff
@@ -0,0 +1,148 @@
+import { equal } from 'node:assert';
+import {
+  Field,
+  Mina,
+  Bytes,
+  Crypto,
+  createEcdsa,
+  createForeignCurve,
+  Hash,
+} from 'o1js';
+
+import {
+  EcdsaProgram,
+  KeccakProgram,
+  CounterProgram,
+  generateEcdsaProof,
+  generateKeccakProof,
+  generateCounterProof,
+} from './sideloaded-zkprograms.js';
+
+import { MyFavoriteZkApp } from './verifier-zkapp.js';
+
+/**
+ * This example showcases the side-loaded proof verification functionality:
+ * 1. Deploys a verifier contract that can verify ECDSA and Keccak proofs
+ * 2. Generates and verifies an ECDSA signature proof
+ * 3. Generates and verifies a Keccak hash preimage proof
+ * 4. Updates contract state after successful verification
+ */
+
+// ---------------------------- Setup Local Blockchain ----------------------------
+const localChain = await Mina.LocalBlockchain({
+  proofsEnabled: false,
+  enforceTransactionLimits: false,
+});
+Mina.setActiveInstance(localChain);
+const fee = 1e8;
+
+// Setup test accounts
+const [deployer, zkAppPrivateKey, alice] = localChain.testAccounts;
+const zkApp = new MyFavoriteZkApp(zkAppPrivateKey.key.toPublicKey());
+
+// ---------------------------- Compile Programs ----------------------------
+console.log('Compiling ZkPrograms and generating verification keys...');
+const ecdsaVerificationKey = (await EcdsaProgram.compile()).verificationKey;
+const keccakVerificationKey = (await KeccakProgram.compile()).verificationKey;
+const counterVerificationKey = (await CounterProgram.compile()).verificationKey;
+const verifierContractKey = (await MyFavoriteZkApp.compile()).verificationKey;
+
+// Log verification key hashes for reference
+console.log('ECDSA verification key hash:', ecdsaVerificationKey.hash.toBigInt());
+console.log('Keccak verification key hash:', keccakVerificationKey.hash.toBigInt());
+console.log('Counter verification key hash:', counterVerificationKey.hash.toBigInt());
+console.log('MyFavoriteZkApp verification key hash:', verifierContractKey.hash.toBigInt());
+
+// Log addresses for reference
+console.log('Deployer Address:', deployer.toBase58());
+console.log('Deployer Address as Field:', deployer.toFields()[0].toBigInt());
+console.log('MyFavoriteZkApp Address:', zkAppPrivateKey.key.toPublicKey().toBase58());
+console.log(
+  'MyFavoriteZkApp Address as Field:',
```

Comment:
nit: same comment about a more descriptive contract name to reduce cognitive over head for first time readers.